store/counter.go

```diff
@@ -0,0 +1,31 @@
+package store
+
+import "sync"
+
+// counter is a simple thread-safe integer.
+type counter struct {
+	sync.RWMutex
+	count int
+}
+
+// increment increments the counter by one.
+func (c *counter) increment() {
+	c.Lock()
+	c.count++
```

Comment:
you can skip out on the locking, and use the atomic package to do the increments.